lib/structuraid_core/elements/reinforcement/straight_longitudinal_layer.rb

```diff
@@ -44,7 +46,15 @@ def reposition(above_middle:, offset: nil)
           offset ||= 0.5 * diameter
 
           [@start_location, @end_location].each do |location|
-            location.value_3 = above_middle ? location.value_3 - offset : location.value_3 + offset
+            location.update_from_matrix(
+              Matrix.column_vector(
+                [
+                  location.value_1,
+                  location.value_2,
+                  above_middle ? location.value_3 - offset : location.value_3 + offset
+                ]
+              )
+            )
```

Comment:
are we still going to use `Matrix` for these 1-dimensional matrices? Shouldn't we replace them with `Vector`?

Reply:
no, here we can implement the Vector, done